main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -281,6 +281,14 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       optional(keyword("with") ~ WS ~ oneOrMore(TypeConstraint).separatedBy(optWS ~ "," ~ optWS)) ~> ((o: Option[Seq[ParsedAst.TypeConstraint]]) => o.getOrElse(Seq.empty))
     }
 
+    def EqualityConstraint: Rule1[ParsedAst.EqualityConstraint] = rule {
+      SP ~ Type ~ optWS ~ "~" ~ optWS ~ Type ~ SP ~> ParsedAst.EqualityConstraint
+    }
+
+    def OptEqualityConstraintList: Rule1[Seq[ParsedAst.EqualityConstraint]] = rule {
+      optional(keyword("where") ~ WS ~ oneOrMore(EqualityConstraint).separatedBy(optWS ~ "," ~ optWS)) ~> ((o: Option[Seq[ParsedAst.EqualityConstraint]]) => o.getOrElse(Seq.empty))
```

Comment:
Hmm.

But, in any case, such eq constraints are likely to be exceedingly rare?

Reply:
Perhaps this would be sufficient:

```
with Eq[a], a ~ b
```

Reply:
But let us worry about it later.

Reply:
Yes this syntax is not meant to be definitive, just enough to get off the ground.